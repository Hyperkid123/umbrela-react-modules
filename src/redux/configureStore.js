import {createStore, applyMiddleware, compose} from 'redux';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import { initialize, setActiveLanguage, addTranslationForLanguage } from 'react-localize-redux';
import en from '../lang/en.js';
import cs from '../lang/cs.js';

const loggerMiddleware = createLogger();
const composeEnhacer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export default function configureStore(preloadedState) {
  const languages = [{
    name: 'Czech',
    code: 'cs',
  },{
    name: 'English',
    code: 'en',
  }];
  let store;
  if(process.env.NODE_ENV !== 'production') {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
          loggerMiddleware
        )
      )
    )
  } else {
    store = createStore(
      rootReducer,
      composeEnhacer(
        applyMiddleware(
          thunkMiddleware,
        ),
      )
    )
  }
  store.dispatch(initialize(languages))
  store.dispatch(addTranslationForLanguage(en, 'en'));
  store.dispatch(addTranslationForLanguage(cs, 'cs'));
  store.dispatch(setActiveLanguage(window.lang));
  return store;
}
