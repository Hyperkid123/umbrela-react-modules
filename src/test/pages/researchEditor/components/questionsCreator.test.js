import React from 'react'
import QuestionsCreator, {QuestionsCreator as Snapshot} from '../../../../pages/researchEditorModule/components/questionsCreator';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('Questions Creator component', () => {
    const initialState  = {
      questions: {
        activeQuestion: {
          questionId: 1
        }
      },
      editor: {
        activeSheet: {
          sheetId: 0,
          title: 'foo'
        }
      },
      locale: {
        languages: [{
          name: "English",
          code: 'en',
          active: true,
        }],
        translations: {}
      }
    };
    const mockStore = configureStore();
    it('should create component correctly', () => {
      let store = mockStore(initialState);
      let wrapper = shallow(<QuestionsCreator store={store}/>);
      expect(wrapper).toBeDefined();
    });

    it('should render correctly', () => {
      const tree = shallow(
        <Snapshot
          translate={jest.fn()}
        />
      ).dive();
      expect(toJson(tree)).toMatchSnapshot();
    });

    it('should close menu', () => {
      const wrapper = shallow(
        <Snapshot
          translate={jest.fn()}
        />
      );
      wrapper.instance().handleRequestClose();
      expect(wrapper.state().open).toBeFalsy();
    });

    it('should open menu', () => {
      const wrapper = shallow(
        <Snapshot
          translate={jest.fn()}
        />
      );
      wrapper.instance().handleClick({currentTarget: 'foo', preventDefault: jest.fn()});
      expect(wrapper.state().open).toBeTruthy();
      expect(wrapper.state().anchorEl).toEqual('foo');
    });

    it('should call new question', () => {
      const createNewQuestion = jest.fn();
      const wrapper = shallow(
        <Snapshot
          translate={jest.fn()}
          createNewQuestion={createNewQuestion}
        />
      );
      wrapper.instance().handleCreateNewQuestion('CloseQuestion');
      expect(createNewQuestion.mock.calls.length).toEqual(1);
    })
});
