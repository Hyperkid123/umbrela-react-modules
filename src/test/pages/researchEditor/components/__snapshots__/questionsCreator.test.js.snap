// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Questions Creator component should render correctly 1`] = `
<div
  className="sc-bdVaJa eVCzwl"
>
  <styled.div>
    <WithStyles
      onClick={[Function]}
    >
      <ShouldUpdate />
    </WithStyles>
    <WithStyles
      onClose={[Function]}
      open={false}
    >
      <WithStyles
        key="0"
        onClick={[Function]}
      />
      <WithStyles
        key="1"
        onClick={[Function]}
      />
      <WithStyles
        key="2"
        onClick={[Function]}
      />
      <WithStyles
        key="3"
        onClick={[Function]}
      />
      <WithStyles
        key="4"
        onClick={[Function]}
      />
      <WithStyles
        key="5"
        onClick={[Function]}
      />
    </WithStyles>
  </styled.div>
  <styled.div
    column={true}
  >
    <Connect(QuestionMenu) />
  </styled.div>
</div>
`;
